topic-operator/src/main/java/io/strimzi/operator/topic/ZkTopicsWatcher.java

```diff
@@ -84,8 +84,11 @@ void start(Zk zk) {
                     LogContext logContext = LogContext.zkWatch(TOPICS_ZNODE, "-" + topicName);
                     topicOperator.onTopicDeleted(logContext, new TopicName(topicName)).setHandler(ar -> {
                         if (ar.succeeded()) {
+                            topicOperator.decrementTopicCounter();
+                            topicOperator.incrementSuccessfulReconciliationsCounter();
                             LOGGER.debug("{}: Success responding to deletion of topic {}", logContext, topicName);
                         } else {
+                            topicOperator.incrementFailedReconciliationsCounter();
```

Comment:
With these I wonder if it would be better to build this into the `Reconciliation` class, rather than having to change all these call sites. Wdyt?

Reply:
Maybe it would. Should I do that in this PR?

Reply:
By all means give it a try and see how it works out. You can always revert/drop that commit.